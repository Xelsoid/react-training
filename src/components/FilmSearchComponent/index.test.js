import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import MOCKED_DATA from '@mockedData/MOCKED_DATA';
import { BrowserRouter } from 'react-router-dom';
import ComponentExample from './index';

const mockStore = configureStore([]);

describe('FilmSearchComponent test', () => {
  let store;

  beforeEach(() => {
    fetchMock.doMock();
    store = mockStore({
      moviesData: MOCKED_DATA,
    });
  });

  it('renders correctly', () => {
    const component = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ComponentExample />
        </BrowserRouter>
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
