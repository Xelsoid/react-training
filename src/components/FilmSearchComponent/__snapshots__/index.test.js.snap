// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmSearchComponent test renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(FilmSearchComponent)>
        <FilmSearchComponent
          addMovies={[Function]}
          match={Object {}}
        >
          <div
            className="film-search-wrapper"
          >
            <h2
              className="film-search__title"
            >
              FIND YOUR MOVIE
            </h2>
            <div
              className="film-search__panel"
            >
              <TextInput
                defaultValue=""
                onChangeCallback={[Function]}
                placeholder="Enter the data"
                type="text"
              >
                <input
                  className="text-input"
                  onChange={[Function]}
                  placeholder="Enter the data"
                  type="text"
                  value=""
                />
              </TextInput>
              <Link
                to="/search/Search  title"
              >
                <LinkAnchor
                  href="/search/Search  title"
                  navigate={[Function]}
                >
                  <a
                    href="/search/Search  title"
                    onClick={[Function]}
                  >
                    <Button
                      additionalClassName="btn--large"
                      name="SEARCH"
                      onClickCallback={[Function]}
                      type="button"
                    >
                      <button
                        className="btn btn--large"
                        onClick={[Function]}
                        type="button"
                      >
                        SEARCH
                      </button>
                    </Button>
                  </a>
                </LinkAnchor>
              </Link>
            </div>
            <div
              className="film-search__filter"
            >
              <span
                className="film-search__filter-title"
              >
                SEARCH BY
              </span>
              <OptionChooser
                defaultValue="title"
                onChangeCallback={[Function]}
                optionsConfig={
                  Array [
                    Object {
                      "name": "searchType",
                      "radioBtnName": "TITLE",
                      "value": "title",
                    },
                    Object {
                      "name": "searchType",
                      "radioBtnName": "GENRE",
                      "value": "genres",
                    },
                  ]
                }
              >
                <div
                  className="option-chooser-wrapper"
                >
                  <input
                    checked={true}
                    className="option-chooser__option"
                    id="title"
                    name="searchType"
                    onChange={[Function]}
                    type="radio"
                    value="title"
                  />
                  <label
                    className="option-chooser__label"
                    htmlFor="title"
                  >
                    TITLE
                  </label>
                  <input
                    checked={false}
                    className="option-chooser__option"
                    id="genres"
                    name="searchType"
                    onChange={[Function]}
                    type="radio"
                    value="genres"
                  />
                  <label
                    className="option-chooser__label"
                    htmlFor="genres"
                  >
                    GENRE
                  </label>
                </div>
              </OptionChooser>
            </div>
          </div>
        </FilmSearchComponent>
      </Connect(FilmSearchComponent)>
    </Router>
  </BrowserRouter>
</Provider>
`;
